crowbar_framework/lib/crowbar/backup/export.rb

```diff
@@ -0,0 +1,194 @@
+
+          Chef::DataBag.load(name).each do |item, item_url|
+            next if item.match(/\Atemplate-.*/)
+            logger.debug "Backing up databag #{name}/#{item}"
+
+            record = Chef::DataBagItem.load(
+              name,
+              item
+            )
+
+            bag_dir.join("#{item}.json").open("w") do |file|
+              file.write(
+                JSON.pretty_generate(
+                  record
+                )
+              )
+            end
+          end
+        end
+      end
+
+      def db
+        logger.debug "Backing up database"
+
+        data_dir = workdir.join("crowbar")
+        data_dir.mkpath
+
+        SerializationHelper::Base.new(
+          YamlDb::Helper
+        ).dump(
+          data_dir.join(
+            "database.#{YamlDb::Helper.extension}"
+          )
+        )
+      end
+
+      def crowbar
+        logger.debug "Backing up Crowbar files"
+
+        data_dir = workdir.join("crowbar")
+        ["keys", "configs", "root"].each do |folder|
+          absolute_path = data_dir.join(folder)
+          absolute_path.mkpath unless absolute_path.directory?
+        end
+
+        self.class.export_files.each do |filemap|
+          source, destination = filemap
+          if source =~ /resolv.conf/
+            data_dir.join(destination).open("w") do |file|
+              forwarders.each do |forwarder|
+                file.write("nameserver #{forwarder}")
+              end
+            end
+          else
+            # copy files with higher permissions
+            dest = data_dir.join(destination).to_s
+            system("sudo", "cp", "-a", source, dest)
+            system("sudo", "chown", "-R", "crowbar:crowbar", dest)
+          end
+        end
+      end
+
+      def meta
+        meta = Hash.new
+        meta["version"] = ENV["CROWBAR_VERSION"]
+        meta["created_at"] = Time.zone.now.to_s
+        meta["platform"] = NodeObject.admin_node.target_platform
+
+        workdir.join("meta.yml").open("w") do |file|
+          file.write(meta.to_yaml)
+        end
+      end
+
+      protected
+
+      def forwarders
+        f = File.open("/etc/bind/named.conf")
+        arr = []
+        write = false
+        f.each_line do |line|
+          if line =~ /forwarders {/
+            write = true
+            next
+          end
+          write = false if write && line =~ /};/
+          arr.push(line) if write
+        end
+        arr.map(&:chomp!).map(&:strip!)
+        arr.each { |s| s.slice!(";") }
+      end
+
+      def workdir
+        @workdir ||= Pathname.new(
+          path
+        )
+      end
+
+      def chef(component, klass)
+        logger.debug "Backing up #{component.pluralize}"
+
+        data_dir = workdir.join("knife", component.pluralize)
+        data_dir.mkpath
+
+        klass.list.each do |name, url|
+          logger.debug "Backing up #{component} #{name}"
+
+          record = klass.load(name)
+
+          unless record
```

Comment:
As discussed earlier, it's unclear when this would happen. If we keep it, then we should at least be consistent and keep it in `databags` too. But I'd vote for dropping it.